ads/google/a4a/utils.js

```diff
@@ -357,92 +356,71 @@ export function additionalDimensions(win, viewportSize) {
 /**
  * Extracts configuration used to build amp-analytics element for active view.
  *
+ * @param {!../../../extensions/amp-a4a/0.1/amp-a4a.AmpA4A} a4a
  * @param {!../../../src/service/xhr-impl.FetchResponseHeaders} responseHeaders
  *   XHR service FetchResponseHeaders object containing the response
  *   headers.
- * @param {!../../../src/service/extensions-impl.Extensions} extensions
- * @return {?AmpAnalyticsConfigDef} config or null if invalid/missing.
+ * @return {?JSONType} config or null if invalid/missing.
  */
-export function extractAmpAnalyticsConfig(responseHeaders, extensions) {
+export function extractAmpAnalyticsConfig(a4a, responseHeaders) {
   if (responseHeaders.has(AMP_ANALYTICS_HEADER)) {
     try {
       const analyticsConfig =
         JSON.parse(responseHeaders.get(AMP_ANALYTICS_HEADER));
-      dev().assert(Array.isArray(analyticsConfig['url']));
-      if (analyticsConfig.url.length) {
-        extensions.loadExtension('amp-analytics');
+      console.log('anayticsConfig is ', analyticsConfig['url']);
+      dev().assert(Array.isArray(analyticsConfig['url']) &&
```

Comment:
length check was not previously within assert, please move it out or upgrade to user assert to ensure it is compiled into production build.  Don't want to create amp-analytics element with no urls